plenoptic/synthesize/eigendistortion.py

```diff
@@ -374,170 +386,51 @@ def _synthesize_power(self, l=0, tol=1e-10, n_steps=1000):
-
-        .. math::
-            \begin{align}
-            A &\approx  Q_kT_kQ_K^T\\
-            \text{where } T_{k}&=\left[\begin{array}{cccc}{\alpha_{1}} & {\beta_{2}} & {} & {} \\ {\beta_{2}} & {\ddots} & {\ddots} & {} \\ {} & {\ddots} & {\ddots} & {\beta_{k}} \\ {} & {} & {\beta_{k}} & {\alpha_{k}}\end{array}\right]\\
-            Q_k &= [\bf{q_1}, \bf{q_2}, ..., \bf{q_k}].
-            \end{align}
-
-        Here :math:`T_k` is a :math:`(k \times k)` tri-diagonal matrix, and :math:`Q_k` is an :math:`(n\times k)` matrix of
-        orthogonal vectors spanning the k-dimensional Krylov subspace of :math:`A`. The matrix :math:`T_k` is easier to
-        diagonalize due to its reduced size and tridiagonal structure, :math:`T_k=V\Lambda V^T`. By subbing this into our
-        earlier approximation for :math:`A`, we find
-
-        .. math::
-            \begin{align}
-            A &\approx Q_kT_kQ_k^T\\
-              &= Q_kV\Lambda V^T Q_k^T
-            \end{align}
-
-        Thus, :math:`Q_kV` are our estimates of :math:`k` eigenvectors of :math:`A` and :math:`\text{diag}(\Lambda )` are
-        their associated eigenvalues.
-
+    def _synthesize_randomized_svd(self, k, p=0, q=0):
+        r"""  Synthesize eigendistortions using randomized truncated SVD.
         Parameters
         ----------
-        n_steps: int
-            number of power iteration steps (i.e. eigenvalues/eigenvectors to compute and/or return). Recommended to do many
-            more than amount of requested eigenvalue/eigenvectors, N. Some say 2N steps but as many as possible is preferred.
-        e_vecs: int or iterable, optional
-           Eigenvectors to return. If num_evecs is an int, it will return all eigenvectors in range(num_evecs).
-           If an iterable, then it will return all vectors selected.
-        debug_A: torch.Tensor
-            For debugging purposes. Explicit FIM for matrix vector multiplication. Bypasses matrix-vector product with
-            implicitly stored FIM of model.
+        k: int
+            Rank of factorization to be returned.
+        p: int, Optional
+            Oversampling parameter.
+        q: int, Optional
+            Matrix power iteration. Used to squeeze the eigen spectrum for more accurate approximation.
 
         Returns
         -------
-        eig_vals: torch.Tensor
-            Tensor of eigenvalues, sorted in descending order. torch.Size(n_steps,) if e_vecs is None.
-            torch.Size(len(e_vecs),) if e_vecs is not None. eigenvalues corresponding to the eigenvectors at that index.
-        eig_vecs: toch.Tensor
-            Tensor of n_steps eigenvectors with torch.Size(n, len(e_vecs)) . If ``return_evecs=False``,
-            then returned  eig_vecs is an empty tensor.
-        eig_vecs_ind: torch.Tensor
-            Indices of each returned eigenvector
+        S: torch.Tensor
+            Eigenvalues, Size((n, )).
+        V: torch.Tensor
+            Eigendistortions, Size((n, k)).
```

Comment:
Also returns `error_approx`, should be documented

Reply:
done.